app/scripts/modules/kubernetes/pipeline/stages/runJob/runJobStage.js

```diff
@@ -23,15 +23,39 @@ module.exports = angular.module('spinnaker.core.pipeline.stage.kubernetes.runJob
       validators: [
         { type: 'requiredField', fieldName: 'account' },
         { type: 'requiredField', fieldName: 'namespace' },
-        { type: 'requiredField', fieldName: 'container.imageDescription.tag', fieldLabel: 'tag' },
+        { type: 'custom', fieldName: 'container.imageDescription.tag', fieldLabel: 'tag', validate: (_pipeline, stage) => {
+          let response = null;
+
+          if (stage.container.imageDescription.fromTrigger === false) {
+            response = '<strong>Tag</strong> is a required field for Run Job stages.';
+          } else if (stage.container.imageDescription.fromTrigger === true && !stage.container.imageDescription.registry ) {
+            response = '<strong>Trigger</strong> is a required field for Run Job stages.';
+          }
+
+          return response;
+        }},
       ]
     });
   }).controller('kubernetesRunJobStageCtrl', function($scope, accountService) {
+
     this.stage = $scope.stage;
+    this.pipeline = $scope.pipeline;
+    this.container = null;
+
     if (!_.has(this.stage, 'container.name')) {
       _.set(this.stage, 'container.name', Date.now().toString());
     }
 
+    if (!_.has(this.stage, 'container.imageDescription.fromTrigger')) {
+      _.set(this.stage, 'container.imageDescription.fromTrigger', false);
+    }
+
+
+    if(this.stage.container.imageDescription.fromTrigger === true) {
+      const imageDescription = this.stage.container.imageDescription;
+      this.container = `${imageDescription.registry}/${imageDescription.organization}/${imageDescription.repository}`;
```

Comment:
I was trying to keep it inline with what the serverGroup configuration does. That selector displays `registry/repository/image:tag`. It looks confusing, but the structure of triggers is different than `imageDescription` and puts `repository/image` in the `repository` field.